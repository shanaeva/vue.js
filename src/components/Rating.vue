<template>
  <h2 class="circle">
    {{ value }}
  </h2>
</template>

<script>
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'Rating',
  props: { value: { type: String, require: true, default: '' } },
});

</script>

<style scoped>
  .circle {
    width: 70px;
    height: 70px;
    border: 1px solid #ffffff;
    color: #a1e66f;
    border-radius: 50%;
    text-align: center;
    line-height: 70px;
  }
</style>
