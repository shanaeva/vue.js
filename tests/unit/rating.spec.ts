import { shallowMount } from '@vue/test-utils';
import Rating from '@/components/Rating.vue';

describe('Rating', () => {
  it('render', () => {
    const wrapper = shallowMount(Rating, { props: { value: 7 } });
    expect(wrapper.props().value).toBe(7);
  });
  it('render correctly', () => {
    const wrapper = shallowMount(Rating, { props: { value: 7 } });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
