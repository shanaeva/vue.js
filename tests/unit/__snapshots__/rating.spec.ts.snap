// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating render correctly 1`] = `
<h2
  class="rating"
>
  7
</h2>
`;
